File: src/main/java/no/nav/fo/veilarbregistrering/enhet/Kommunenummer.java
package no.nav.fo.veilarbregistrering.enhet;

import java.util.Objects;

public class Kommunenummer {

    private final String kommunenummer;

    public static Kommunenummer of(String kommunenummer) {
        return new Kommunenummer(kommunenummer);
    }

    public static Kommunenummer of(KommuneMedBydel kommuneMedBydel) {
        return new Kommunenummer(kommuneMedBydel.kommenummer);
    }

    private Kommunenummer(String kommunenummer) {
        this.kommunenummer = kommunenummer;
    }

    public String asString() {
        return kommunenummer;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Kommunenummer that = (Kommunenummer) o;
        return Objects.equals(kommunenummer, that.kommunenummer);
    }

    @Override
    public int hashCode() {
        return Objects.hash(kommunenummer);
    }

    public boolean kommuneMedBydeler() {
        return KommuneMedBydel.contains(kommunenummer);
    }

}

File: src/main/java/no/nav/fo/veilarbregistrering/enhet/Kommunenummer.kt
package no.nav.fo.veilarbregistrering.enhet

data class Kommunenummer (val kommunenummer: String) {

    fun kommuneMedBydeler(): Boolean = kommunenummer in KommuneMedBydel

    companion object {
        fun of(kommunenummer: String): Kommunenummer {
            return Kommunenummer(kommunenummer)
        }

        internal fun of(kommuneMedBydel: KommuneMedBydel): Kommunenummer {
            return Kommunenummer(kommuneMedBydel.kommenummer)
        }
    }
}